tests/ord-service/tests/api_test.go

```diff
@@ -471,18 +483,48 @@ func TestORDService(t *testing.T) {
 }
 
 func makeRequest(t *testing.T, httpClient *http.Client, url string) string {
-	return makeRequestWithStatusExpect(t, httpClient, url, http.StatusOK)
+	return makeRequestWithHeadersAndStatusExpect(t, httpClient, url, map[string][]string{}, http.StatusOK)
+}
+
+func makeRequestWithHeaders(t *testing.T, httpClient *http.Client, url string, headers map[string][]string) string {
+	return makeRequestWithHeadersAndStatusExpect(t, httpClient, url, headers, http.StatusOK)
 }
 
 func makeRequestWithStatusExpect(t *testing.T, httpClient *http.Client, url string, expectedHTTPStatus int) string {
+	return makeRequestWithHeadersAndStatusExpect(t, httpClient, url, map[string][]string{}, expectedHTTPStatus)
+}
+
+func makeRequestWithHeadersAndStatusExpect(t *testing.T, httpClient *http.Client, url string, headers map[string][]string, expectedHTTPStatus int) string {
 	request, err := http.NewRequest(http.MethodGet, url, nil)
 	require.NoError(t, err)
 
+	for key, values := range headers {
+		for _, value := range values {
+			request.Header.Add(key, value)
+		}
+	}
+
 	response, err := httpClient.Do(request)
 
 	require.NoError(t, err)
 	require.Equal(t, expectedHTTPStatus, response.StatusCode)
 
+	if !strings.Contains(url, "/specification") {
+		reqFormatPattern := regexp.MustCompile(fmt.Sprintf("^.*\\$format=(.*)$"))
+		matches := reqFormatPattern.FindStringSubmatch(url)
```

Comment:
It would be easier to read if you parse the URL and then get the query with the URL utils